### ISSUE TITLE
Failure of unicode sandwich for multi-dimensional columns

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#9614]
We allow comparisons of unicode with byte-valued columns using a unicode sandwich, but this fails in the corner-case where the column is multidimensional and we take a single element from it:
```
from astropy.table import Column
c = Column([[b'a', b'b'], [b'c', b'd']], name='c')
print(c == ['a', 'b'])
# [[ True  True]
#  [False False]]
print(c[0] == ['a', 'b'])
# False  # expected [True, True]
```

The reason is that for multi-D columns, access of any given `Column` row just returns a plain `ndarray`, which does not know about the unicode sandwich: https://github.com/astropy/astropy/blob/master/astropy/table/_column_mixins.pyx#L54-L55

A fix might be to convert the result to unicode or to use an `ndarray` subclass that still contains the unicode sandwich (but is otherwise a plain `ndarray`).

p.s. Found while trying a new `Masked` class for creating a new `MaskedColumn`.

**Comment by taldcroft** [comment]
I suspect this is a bit of a rabbit hole...  with new numpy dtypes is there hope for a single-byte character type anywhere on the horizon before I'm retired?

**Comment by mhvk** [comment]
Yes, hope, but before you're retired... that may depend on how long you plan to continue working....

**Comment by ngoldbaum** [comment]
> is there hope for a single-byte character type anywhere on the horizon before I'm retired?

@taldcroft for what it's worth, we have both a real ASCII and Bytes DType based on the UTF-8 String DType in NumPy 2.0 in a NASA ROSES proposal we just put in. Fingers' crossed that gets funded. With StringDType present in NumPy now IMO it's much less technically difficult to solve this problem.

Also like to screenshot your comment in this thread above from 2020 and use it in the SciPy talk I'm giving on StringDType. Of course StringDType doesn't fix this precise issue but your comment illustrates in a fun, if bittersweet way, the sort of pain and technical debt lack the current numpy string situation has caused. Please let me know if you'd prefer I not talk about astropy and the unicode sandwich in my talk.
